@kiva/kv-shop/src/useBraintreeDropIn.ts

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import numeral from 'numeral';
+import { ref } from 'vue-demi';
+import type { ApplePayPaymentRequest } from 'braintree-web';
+import type { Dropin } from 'braintree-web-drop-in';
+
+export type PaymentType = 'card' | 'paypal' | 'paypalCredit' | 'venmo' | 'applePay' | 'googlePay';
+export type PayPalFlowType = 'checkout' | 'vault';
+
+export const defaultPaymentTypes = ['paypal', 'card', 'applePay', 'googlePay'] as const;
+
+export interface DropInInitOptions {
+	amount: string|number,
+	container: string|HTMLElement,
+	googlePayMerchantId: string,
+	paymentTypes?: PaymentType[],
+	preselectVaultedPaymentMethod?: boolean,
+	paypalFlow?: PayPalFlowType,
+}
+
+export default function useBraintreeDropIn() {
+	let instance: Dropin;
+	let formattedAmount = '';
+	const paymentMethodRequestable = ref(false);
+
+	function getApplePaymentRequest(amount: string): ApplePayPaymentRequest {
+		// @ts-ignore
+		return {
+			countryCode: 'US',
+			currencyCode: 'USD',
+			// merchantCapabilities: ['supports3DS'], // TODO: confirm/update
+			requiredBillingContactFields: ['postalAddress'],
+			// supportedNetworks: ['amex', 'discover', 'interac', 'jcb', 'masterCard', 'visa'], // TODO: confirm/update
```

Comment:
Which credit card networks do we actually accept? I know that we don't accept all the networks because we see errors like this crop up: "Invalid request: TRANSACTION_PAYMENT_METHOD_NONCE_CARD_TYPE_IS_NOT_ACCEPTED: Payment instrument type is not accepted by this merchant account."

I had to `@ts-ignore` several lines in the Dropin config because these are required values but somehow we've been using the Dropin so far without defining these options.

Reply:
Did this mark or come along with an update to the version of BT we're using?

In general, the interface had previously attempted to determine the card type based on input and we'd just let any errors on submission bubble up. I think Leo would know about the backend handling for exact payment types that we do or do not accept.

I guess if we "have" to start adding this config we can with the benefit it would likely provide more immediate feedback in the BT dropin interface to the user!

Reply:
The BT version in UI is 1.30.0, while the version here is 1.37.0, though I don't see anything in the changelogs about supportedNetworks being newly required.

The Dropin still initializes fine, so I think the options aren't actually required for the BT Dropin (the requirements come from the types as defined by Apple and Google, so BT might be applying default options when those aren't specified).

I think defining these options would be more like a UX enhancement.